src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -0,0 +1,164 @@
+import 'reflect-metadata';
+import 'chai/register-should';
+import { HttpRequestExecutor } from './HttpRequestExecutor';
+import {
+  DefaultVirtualScripts,
+  VirtualScript,
+  VirtualScripts,
+  VirtualScriptType
+} from '../Scripts';
+import { Protocol } from './Protocol';
+import { RequestExecutor } from './RequestExecutor';
+import { Request, RequestOptions } from './Request';
+import { RequestExecutorOptions } from './RequestExecutorOptions';
+import { expect } from 'chai';
+import nock from 'nock';
+import {
+  anyString,
+  anything,
+  instance,
+  mock,
+  reset,
+  spy,
+  verify,
+  when
+} from 'ts-mockito';
+import { container, Lifecycle } from 'tsyringe';
+import { URL } from 'url';
+
+describe('HttpRequestExecutor', () => {
+  const virtualScriptsMock = mock<VirtualScripts>(DefaultVirtualScripts);
+
+  let requestOptions: RequestOptions;
+  let request: Request;
+  let spiedRequest: Request;
+  let requestExecutorOptions: RequestExecutorOptions;
+
+  beforeEach(() => {
+    requestOptions = { url: 'https://foo.bar', headers: {} };
+    request = new Request(requestOptions);
+    spiedRequest = spy(request);
+    when(spiedRequest.method).thenReturn('GET');
```

Comment:
Each test must act on its own data. It would be better to extract that in a separate function. Wdyt?

```ts
const createRequest = (options?: Partial<RequestOptions>) => {
    const request = new Request({ url: 'https://foo.bar', headers: {}, ...(options ?? {}) };);
    spiedRequest = spy(request);
    when(spiedRequest.method).thenReturn('GET');

   return { spiedRequest, request }
}

it('should do something', async () => {
  const { request, spiedRequest } = createRequest({ headers: { testHeader: 'test-header-value' } });
  
  await executor.execute(request);

  verify(spiedRequest.setHeaders(headers)).once();  
});
```

Reply:
`beforeEach` gets executed, well, before each test and hence each test works with its own `request`, `container` etc.
Frankly speaking I see no difference at all but I've changed it to the proposed approach because why not.